finance-tracker/app/controllers/user_stocks_controller.rb

```diff
@@ -1,2 +1,22 @@
 class UserStocksController < ApplicationController
+  def create
+    stock = Stock.find_by(ticker: (params[:ticker]))
+    if stock.blank? || !current_user.stocks.exists?(params[:ticker])
+      stock = Stock.new_lookup(params[:ticker])
+      if stock.save
+        stock.users << current_user
+        flash[:notice] = "Stock #{stock.name} was successfully added to your portfolio"
+      else
+        flash[:notice] = "Stock #{params[:ticker]} could not be added to your portfolio"
+      end
+    end
+    redirect_to portfolio_index_path
+  end
+
+  def destroy
+    stock =  Stock.find(params[:id])
+    stock.user_stocks.destroy_by(user_id: current_user.id)
```

Comment:
```suggestion
    stock =  current_user.stocks.find(params[:id])
    stock.destroy
```
Can you try the above? If its wrong, then I think its better if we do:
```
    stock =  current_user.user_stocks.find(params[:id])
    stock.destroy
```